docs/src/pages/[platform]/connected-components/in-app-messaging/index.page.mdx

```diff
@@ -0,0 +1,86 @@
+supportedFrameworks: react|react-native
+---
+
+import { Fragment } from '@/components/Fragment';
+import { getCustomStaticPath } from "@/utils/getCustomStaticPath";
+
+import { 
+  IN_APP_MESSAGE_DISPLAY_PROPS, 
+  MESSAGE_COMPONENTS, 
+  ReactPropsTable,
+  USE_IN_APP_MESSAGING_API, 
+} from './props';
+
+export async function getStaticPaths() {
+  return getCustomStaticPath(frontmatter.supportedFrameworks);
+}
+
+{/*  `getStaticProps` is required to prevent "Error: getStaticPaths was added without a getStaticProps. Without getStaticProps, getStaticPaths does nothing" */}
+
+export async function getStaticProps() {
+  return { props: {} }
+}
+
+In-App Messaging allows you to better engage users with contextually appropriate messages rendered on events triggered by their activity while using your application. Create messages that look native to your application and deliver them to your users all without additional code changes.
+>Amplify UI In-App Messaging uses 
+[Amplify In-App Messaging API](https://docs.amplify.aws/lib/in-app-messaging/overview/q/platform/react/)
+and [Amazon Pinpoint](https://docs.aws.amazon.com/pinpoint/latest/userguide/channels-inapp.html).
+
+## Getting Started
+
+> Prerequisites: 
+> - [Create an In-App Messaging campaign](https://docs.amplify.aws/lib/in-app-messaging/prerequisites/q/platform/js/)
+> - [install the required dependencies](../getting-started/installation)
+
+## Setup
+
+Integration with your application can be done with the `InAppMessagingProvider` and `InAppMessageDisplay` components directly, 
+or wrap your app in `withInAppMessaging` (a [React Higher-Order Component](https://reactjs.org/docs/higher-order-components.html)):
+
+<Fragment platforms={['react', 'react-native']}>{({ platform }) => import(`./examples/getting-started.${platform}.mdx`)}</Fragment>
+
+>[`InAppMessaging.syncMessages`](https://docs.amplify.aws/lib/in-app-messaging/sync-messages/q/platform/react-native) is an asynchronous function that handles syncing remote in-app messages with the end user application.
+
+## `InAppMessagingProvider` and `useInAppMessaging`
+
+`InAppMessagingProvider` exposes the value of the `InAppMessagingContext` (a [React context](https://reactjs.org/docs/context.html)) 
+to its `children`. `useInAppMessaging` can be used to directly interact with `InAppMessagingContext` from within `InAppMessagingProvider`.
+
+`useInAppMessaging` exposes the following functions and values of the `InAppMessagingContext`:
+
+<ReactPropsTable props={USE_IN_APP_MESSAGING_API} />
+
+In some use cases, you may want to forego the usage of the default UI handling altogether while still leveraging the Amplify provided In-App Messaging React context and provider for in-app message context state. This can be achieved by wrapping your application in an `InAppMessagingProvider` and utilizing the `useInAppMessaging` hook to expose the values of the `InAppMessagingContext`.
```

Comment:
```suggestion
In some use cases, you may want to forgo the usage of the default UI handling altogether while still leveraging the Amplify provided In-App Messaging React context and provider for in-app message context state. This can be achieved by wrapping your application in an `InAppMessagingProvider` and utilizing the `useInAppMessaging` hook to expose the values of the `InAppMessagingContext`.
```